src/commands/coverage/__tests__/upload.test.ts

```diff
@@ -0,0 +1,345 @@
+        config: {},
+        context,
+      })
+      const fileNames = Object.values(result).flatMap((paths) => paths)
+
+      expect(fileNames.length).toEqual(1)
+
+      expect(fileNames[0]).toEqual('./src/commands/coverage/__tests__/fixtures/jacoco-report.xml')
+    })
+    test('should not fail for invalid single files', async () => {
+      const context = createMockContext()
+      const command = new UploadCodeCoverageReportCommand()
+      const result = command['getMatchingCoverageReportFilesByFormat'].call({
+        basePaths: ['./src/commands/coverage/__tests__/fixtures/does-not-exist.xml'],
+        config: {},
+        context,
+      })
+
+      const fileNames = Object.values(result).flatMap((paths) => paths)
+
+      expect(fileNames.length).toEqual(0)
+    })
+    test('should allow folder and single unit paths', async () => {
+      const context = createMockContext()
+      const command = new UploadCodeCoverageReportCommand()
+      const result = command['getMatchingCoverageReportFilesByFormat'].call({
+        basePaths: [
+          './src/commands/coverage/__tests__/fixtures',
+          './src/commands/coverage/__tests__/fixtures/subfolder/nested-jacoco-report.xml',
+        ],
+        config: {},
+        context,
+      })
+
+      const fileNames = Object.values(result).flatMap((paths) => paths)
+
+      expect(fileNames[0]).toEqual('./src/commands/coverage/__tests__/fixtures/another-jacoco-report.xml')
+      expect(fileNames[1]).toEqual('./src/commands/coverage/__tests__/fixtures/jacoco-report.xml')
+      expect(fileNames[2]).toEqual('./src/commands/coverage/__tests__/fixtures/subfolder/nested-jacoco-report.xml')
+    })
+    test('should not have repeated files', async () => {
+      const context = createMockContext()
+      const command = new UploadCodeCoverageReportCommand()
+      const result = command['getMatchingCoverageReportFilesByFormat'].call({
+        basePaths: [
+          './src/commands/coverage/__tests__/fixtures',
+          './src/commands/coverage/__tests__/fixtures/jacoco-report.xml',
+        ],
+        config: {},
+        context,
+      })
+
+      const fileNames = Object.values(result).flatMap((paths) => paths)
+
+      expect(fileNames.length).toEqual(2)
+    })
+    test('should fetch nested folders', async () => {
+      const context = createMockContext()
+      const command = new UploadCodeCoverageReportCommand()
+      const result = command['getMatchingCoverageReportFilesByFormat'].call({
+        basePaths: ['**/coverage/**/*.xml'],
+        config: {},
+        context,
+      })
+
+      const fileNames = Object.values(result).flatMap((paths) => paths)
+
+      expect(fileNames).toEqual([
+        'src/commands/coverage/__tests__/fixtures/another-jacoco-report.xml',
+        'src/commands/coverage/__tests__/fixtures/jacoco-report.xml',
+        'src/commands/coverage/__tests__/fixtures/subfolder/nested-jacoco-report.xml',
+      ])
+    }, 10000)
```

Comment:
the test is actually sync, right? No `await` within the test body. The timeout config will not have any effect then. Also: you may remove the `async` from the test function definition (this also applies to all other tests)

Reply:
Good catch, removed `async` from the tests that have no `await`.